### PR TITLE
document berry.c.

### PR DESCRIPTION
**Pull request by ProjectRevoTPP**

### CODE REVIEW

File: src/field/field_control_avatar.c [hunk 1]
```diff
@@ -856,7 +856,8 @@ u8 sub_8068F18(void)
     return 0;
 }
 
-u8 *GetFieldObjectScriptPointerForComparison(void)
+// GetFieldObjectScriptPointerPlayerFacing
```
Comment (ProjectRevoTPP) [inline src/field/field_control_avatar.c]: oops, need to fix

File: src/field/berry.c [hunk 2]
```diff
@@ -1099,91 +1126,96 @@ void ClearBerryTrees(void)
...
+    for (i = 0; i < MAX_BERRY_TREES; i++)
         saveBlock1->berryTrees[i] = berryTree;
 }
 
-bool32 BerryTreeGrow(struct BerryTree *tree)
+// when the player does not interact with the tree for a period of time, this is called to advance the grow state.
+static bool32 BerryTreeGrow(struct BerryTree *tree)
 {
-    if (tree->growthSparkle)
+    if (tree->growthSparkle != FALSE)
```
Comment (huderlem) [inline src/field/berry.c]: This looks gross. Is it needed?

File: src/field/berry.c [hunk 3]
```diff
@@ -1099,91 +1126,96 @@ void ClearBerryTrees(void)
...
-            if (minutes >= GetStageDurationByBerryType(tree->berry) * 71)
+            // the player has waited too long to water the berry. Reset the tree. This is because if the berry state is not in the unwatered state, the tree will grow anyway despite this check, which means BerryTreeGrow will handle the regrow process for this, removing the need for this check. This only handles the unwatered soil state.
+            if (totalMinutes >= GetStageDurationByBerryType(tree->berry) * 71)
             {
                 *tree = gBlankBerryTree;
             }
             else
             {
-                s32 time = minutes;
+                // not sure why Game Freak thought a temporary variable was necessary. because time is altered below, perhaps they thought it was unsafe to change it, even though that is not how passed arguments behave.
```
Comment (huderlem) [inline src/field/berry.c]: Unhelpful comment

Comment (ProjectRevoTPP) [reply src/field/berry.c]: i cant get rid of this temp var, so evidently GF put it there for some reason. I don't think I need vague things unexplained.

File: src/field/berry.c [hunk 4]
```diff
@@ -1099,91 +1126,96 @@ void ClearBerryTrees(void)
...
-                        tree->minutesUntilNextStage -= time;
+                        tree->minutesUntilNextStage -= time; // its been X minutes since the last berry update, so update minutesUntilNextStage appropriately to match the time offset that has passed since the update.
                         break;
                     }
+                    // perform the subtraction the other way around to get the number of minutes since the inferred stage update that occured, since minutesUntilNextStage is <= time. we may need this variable to simulate multiple berry cycles in the while loop.
                     time -= tree->minutesUntilNextStage;
-                    tree->minutesUntilNextStage = GetStageDurationByBerryType(tree->berry);
-                    if (!BerryTreeGrow(tree))
+                    tree->minutesUntilNextStage = GetStageDurationByBerryType(tree->berry); // since the tree was inferred to update, set the new minutesUntilNextStage.
+                    if (BerryTreeGrow(tree) == FALSE)
```
Comment (huderlem) [inline src/field/berry.c]: Why is this changed to be explicit?

Comment (ProjectRevoTPP) [reply src/field/berry.c]: It is a boolean check. I would rather code be explicit than someone easily glance over it and make mistakes because a tiny, hard to see "!".

File: src/field/berry.c (hunk 3 above)
Comment (huderlem) [reply src/field/berry.c]: This comment doesn't help anything. Please remove it.
